app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt

```diff
@@ -81,8 +87,36 @@ class MainViewModel(appContext: Application) : AndroidViewModel(appContext) {
     private var getFileDetailsJob = Job()
     private var syncOfflineFilesJob = Job()
 
+    lateinit var uploadFilesHelper: UploadFilesHelper
+
     private fun getContext() = getApplication<MainApplication>()
 
+    fun setCurrentFolder(folder: File?) {
+        folder?.let {
+            it.saveToSavedStateHandle(savedStateHandle)
+            currentFolder.value = it
+        }
+    }
+
+    fun initUploadFilesHelper(fragmentActivity: FragmentActivity, navController: NavController) {
+        uploadFilesHelper = UploadFilesHelper(
+            activity = fragmentActivity,
+            navController = navController,
+            onOpeningPicker = {
+                getContext().trackNewElementEvent("uploadFile")
```

Comment:
either use the context of the viewModel if it works (i'm not sure as it's for a callback)
or directly use the activity instead of a context